spring-integration-cloudevents/src/main/java/org/springframework/integration/cloudevents/transformer/ToCloudEventTransformer.java

```diff
@@ -0,0 +1,288 @@
+		this.cloudEventExtensionExpressions = cloudEventExtensionExpressions;
+	}
+
+	/**
+	 * Construct a ToCloudEventTransformer with no {@link Expression}s for extensions.
+	 *
+	 */
+	public ToCloudEventTransformer() {
+		this((Expression[]) null);
+	}
+
+	/**
+	 * Set the {@link Expression} for creating CloudEvent ids.
+	 * Default expression extracts the id from the {@link MessageHeaders} of the message.
+	 *
+	 * @param idExpression the expression used to create the id for each CloudEvent
+	 */
+	public void setIdExpression(Expression idExpression) {
+		this.idExpression = idExpression;
+	}
+
+	/**
+	 * Set the {@link Expression} for creating CloudEvent source.
+	 * Default expression is {@code "/spring/" + appName + "." + getBeanName())}.
+	 *
+	 * @param sourceExpression the expression used to create the source for each CloudEvent
+	 */
+	public void setSourceExpression(Expression sourceExpression) {
+		this.sourceExpression = sourceExpression;
+	}
+
+	/**
+	 * Set the {@link Expression} for extracting the type for the CloudEvent.
+	 * Default expression sets the default to "spring.message".
+	 *
+	 * @param typeExpression the expression used to create the type for each CloudEvent
+	 */
+	public void setTypeExpression(Expression typeExpression) {
+		this.typeExpression = typeExpression;
+	}
+
+	/**
+	 * Set the {@link Expression} for creating the dataSchema for the CloudEvent.
+	 * Default {@link Expression} evaluates to a null.
+	 *
+	 * @param dataSchemaExpression the expression used to create the dataSchema for each CloudEvent
+	 */
+	public void setDataSchemaExpression(Expression dataSchemaExpression) {
+		this.dataSchemaExpression = dataSchemaExpression;
+	}
+
+	/**
+	 * Set the {@link Expression} for creating the subject for the CloudEvent.
+	 * Default {@link Expression} evaluates to a null.
+	 *
+	 * @param subjectExpression the expression used to create the subject for each CloudEvent
+	 */
+	public void setSubjectExpression(Expression subjectExpression) {
+		this.subjectExpression = subjectExpression;
+	}
+
+	@Override
+	protected void onInit() {
+		super.onInit();
+		this.evaluationContext = ExpressionUtils.createStandardEvaluationContext(getBeanFactory());
+		ApplicationContext applicationContext = getApplicationContext();
+		if (this.sourceExpression == null) {  // in the case the user sets the value prior to onInit.
+			this.sourceExpression = new FunctionExpression<>((Function<Message<?>, URI>) message -> {
+				String appName = applicationContext.getEnvironment().getProperty("spring.application.name");
+				appName = appName == null ? "unknown" : appName;
+				return URI.create("/spring/" + appName + "." + getBeanName());
+			});
+		}
+		if (this.dataSchemaExpression == null) { // in the case the user sets the value prior to onInit.
+			this.dataSchemaExpression = new FunctionExpression<>((Function<Message<?>, @Nullable URI>)
+					message -> null);
```

Comment:
Why cannot we deal with just null for this `dataSchemaExpression`?
No need in a `Function` overhead just to return `null`